### PR TITLE
print plan only if no errors

### PR DESCRIPTION
**Pull request by AviAvni**

### CODE REVIEW

File: src/execution_plan/execution_plan_debug.c [hunk 1]
```diff
@@ -31,6 +32,8 @@ void _ExecutionPlan_Print(const OpBase *op, RedisModuleCtx *ctx, sds *buffer,
 void ExecutionPlan_Print(const ExecutionPlan *plan, RedisModuleCtx *ctx) {
 	ASSERT(plan && ctx);
 
+	if(ErrorCtx_EncounteredError()) return;
```
Comment (swilly22) [inline src/execution_plan/execution_plan_debug.c:35]: Please remove.

In `cmd_explain.c`

```c
ExecutionPlan_PreparePlan(plan);
ExecutionPlan_Init(plan);       // Initialize the plan's ops.
if(ErrorCtx_EncounteredError()) {
    ErrorCtx_EmitException();
    goto cleanup;
}

ExecutionPlan_Print(plan, ctx); // Print the execution plan.
cleanup:
```

**Review by swilly22 — CHANGES_REQUESTED** [review]

File: src/execution_plan/execution_plan_debug.c (hunk 1 above)
Comment (AviAvni) [reply src/execution_plan/execution_plan_debug.c:35]: why?
every time calling to ExecutionPlan_Print it is needed

Comment (swilly22) [reply src/execution_plan/execution_plan_debug.c:35]: this is mixing boundaries, it is preferred that the execution-plan won't be aware of ErrorCtx,
also it seems weird that a call made to `ExecutionPlan_PreparePlan` won't print anything...

**Review by swilly22 — APPROVED** [review]
Looking great!

**Comment by swilly22** [comment]
@AviAvni tests fail due to `redis-py` usage of `hiredis` parser,
https://github.com/andymccurdy/redis-py#parsers
see if for failing test we can initialize `redis-py` without `hiredis`. (`Unhandled exception: can't pickle hiredis.Reader objects`)

**Review by swilly22 — APPROVED** [review]

**Pull request merged**